R/backend-snowflake.R

```diff
@@ -18,13 +18,198 @@ NULL
+      },
+      wday = function(x, label = FALSE, abbr = TRUE, week_start = NULL) {
+        if (!label) {
+          week_start <- week_start %||% getOption("lubridate.week.start", 7)
+          offset <- as.integer(7 - week_start)
+          sql_expr(EXTRACT("dayofweek", DATE(!!x) + !!offset) + 1)
+        } else if (label && !abbr) {
+          sql_expr(
+            DECODE(
+              EXTRACT("dayofweek", !!x),
+              1, "Monday",
+              2, "Tuesday",
+              3, "Wednesday",
+              4, "Thursday",
+              5, "Friday",
+              6, "Saturday",
+              0, "Sunday"
+            )
+          )
+        } else if (label && abbr) {
+          sql_expr(DAYNAME(!!x))
+        } else {
+          abort("Unrecognized arguments to `wday`")
+        }
+      },
+      yday = function(x) sql_expr(EXTRACT("dayofyear", !!x)),
+      week = function(x) {
+        sql_expr(FLOOR((EXTRACT("dayofyear", !!x) - 1L) / 7L) + 1L)
+      },
+      isoweek = function(x) sql_expr(EXTRACT("weekiso", !!x)),
+      month = function(x, label = FALSE, abbr = TRUE) {
+        if (!label) {
+          sql_expr(EXTRACT("month", !!x))
+        } else {
+          if (abbr) {
+            sql_expr(MONTHNAME(!!x))
+          } else {
+            sql_expr(
+              DECODE(
+                EXTRACT("month", !!x),
+                1, "January",
+                2, "February",
+                3, "March",
+                4, "April",
+                5, "May",
+                6, "June",
+                7, "July",
+                8, "August",
+                9, "September",
+                10, "October",
+                11, "November",
+                12, "December"
+              )
+            )
+          }
+        }
+      },
+      quarter = function(x, with_year = FALSE, fiscal_start = 1) {
+        if (fiscal_start != 1) {
+          abort("`fiscal_start` is not supported in Snowflake translation. Must be 1.")
+        }
+
+        if (with_year) {
+          sql_expr((EXTRACT("year", !!x) || "." || EXTRACT("quarter", !!x)))
+        } else {
+          sql_expr(EXTRACT("quarter", !!x))
+        }
+      },
+      isoyear = function(x) {
+        sql_expr(EXTRACT("year", !!x))
+      },
+      seconds = function(x) {
+        build_sql("INTERVAL '", x, " second'")
+      },
+      minutes = function(x) {
+        build_sql("INTERVAL '", x, " minute'")
+      },
+      hours = function(x) {
+        build_sql("INTERVAL '", x, " hour'")
+      },
+      days = function(x) {
+        build_sql("INTERVAL '", x, " day'")
+      },
+      weeks = function(x) {
+        build_sql("INTERVAL '", x, " week'")
+      },
+      months = function(x) {
+        build_sql("INTERVAL '", x, " month'")
+      },
+      years = function(x) {
+        build_sql("INTERVAL '", x, " year'")
+      },
+      # https://docs.snowflake.com/en/sql-reference/functions/date_trunc.html
+      floor_date = function(x, unit = "seconds") {
```

Comment:
The default is plural, below is always singular. I see this is also the case for the postgres backend...
Either you simply fix the default to `"seconds"` or if you want to be more user friendly you could also allow the plural version. Up to you.

Reply:
Good catch, maybe I'll follow up with an update to the postgres backend.  Since the singular and plural versions are all supported on both Snowflake ([here](https://docs.snowflake.com/en/sql-reference/functions-date-time.html#label-supported-date-time-parts)) and lubridate, I'm expanding the arg list to include both.